cairo/ethereum/utils/bytes.cairo

```diff
@@ -0,0 +1,58 @@
+from ethereum_types.bytes import Bytes, BytesStruct
+from ethereum_types.numeric import bool
+from ethereum.utils.numeric import is_zero
+from starkware.cairo.common.math import assert_not_equal
+
+func Bytes__eq__(_self: Bytes, other: Bytes) -> bool {
+    if (_self.value.len != other.value.len) {
+        tempvar res = bool(0);
+        return res;
+    }
+
+    // Case diff: we can let the prover do the work of iterating over the bytes,
+    // return the first different byte index, and assert in cairo that the a[index] != b[index]
+    tempvar is_diff;
+    tempvar diff_index;
+    %{
+        self_bytes = b''.join([memory[ids._self.value.data + i].to_bytes(1, "little") for i in range(ids._self.value.len)])
+        other_bytes = b''.join([memory[ids.other.value.data + i].to_bytes(1, "little") for i in range(ids.other.value.len)])
+        diff_index = next((i for i, (b_self, b_other) in enumerate(zip(self_bytes, other_bytes)) if b_self != b_other), None)
+        if diff_index is not None:
+            ids.is_diff = 1
+            ids.diff_index = diff_index
+        else:
+            # No differences found in common prefix. Lengths were checked before
+            ids.is_diff = 0
+            ids.diff_index = 0
+    %}
+
+    if (is_diff == 1) {
+        // Assert that the bytes are different at the first different index
+        assert_not_equal(_self.value.data[diff_index], other.value.data[diff_index]);
```

Comment:
should we add comments where we add code that can trigger a cairo error over an exceptional halt etc.?
this is equivalent to a panic